File: 01_Language/01_Functions/python/addslashes.py
# coding: utf-8

def addslashes(s):
    s = s.replace('\\', '\\\\') \
        .replace("'", "\\'") \
        .replace('"', '\\"') \
        .replace('\0', '\\0')
    return s


if __name__ == '__main__':
    s = "' \" \\ \0"
    print(s)
    print(addslashes(s))

File: 01_Language/01_Functions/js/addslashes.js
function addslashes(str) {
    return str.replace('\\', '\\\\')
        .replace("'", "\\'")
        .replace('"', '\\"')
        .replace('\0', '\\0')
}

let str = "' \" \\ \0"
console.log(str)
console.log(addslashes(str))
